odoo/addons/test_read_group/tests/test_fill_temporal.py

```diff
@@ -0,0 +1,746 @@
+
+    This feature is mainly used in graph view. For more informations, read the
+    documentation of models's '_read_group_fill_temporal' method.
+    """
+
+    def setUp(self):
+        super(TestFillTemporal, self).setUp()
+        self.Model = self.env['test_read_group.fill_temporal']
+
+    def test_date_range_and_flag(self):
+        """Simple date range test, the flag is also tested.
+
+        One of the most simple test. It must verify that dates 'holes' are filled
+        only when the fill_temporal flag is set.
+        """
+        self.Model.create({'date': '1916-08-18', 'value': 2})
+        self.Model.create({'date': '1916-10-19', 'value': 3})
+        self.Model.create({'date': '1916-12-19', 'value': 5})
+
+        groups = self.Model.with_context(fill_temporal=False).read_group([],
+                                                                         fields=['date',
+                                                                                 'value'],
+                                                                         groupby=['date'])
+
+        expected = [{'__domain': ['&',
+                                  ('date', '>=', '1916-08-01'),
+                                  ('date', '<', '1916-09-01')],
+                     'date': 'August 1916',
+                     'date_count': 1,
+                     'value': 2},
+                    {'__domain': ['&',
+                                  ('date', '>=', '1916-09-01'),
+                                  ('date', '<', '1916-10-01')],
+                     'date': 'September 1916',
+                     'date_count': 0,
+                     'value': False},
+                    {'__domain': ['&',
+                                  ('date', '>=', '1916-10-01'),
+                                  ('date', '<', '1916-11-01')],
+                     'date': 'October 1916',
+                     'date_count': 1,
+                     'value': 3},
+                    {'__domain': ['&',
+                                  ('date', '>=', '1916-11-01'),
+                                  ('date', '<', '1916-12-01')],
+                     'date': 'November 1916',
+                     'date_count': 0,
+                     'value': False},
+                    {'__domain': ['&',
+                                  ('date', '>=', '1916-12-01'),
+                                  ('date', '<', '1917-01-01')],
+                     'date': 'December 1916',
+                     'date_count': 1,
+                     'value': 5}]
+
+        self.assertEqual(groups, [group for group in expected if group['date_count']])
+
+        groups = self.Model.with_context(fill_temporal=True).read_group([],
+                                                                        fields=['date',
+                                                                                'value'],
+                                                                        groupby=['date'],)
```

Comment:
Readability: indenting around column 60 is not good. You spend 4 lines for two simple chained calls. I suggest introducing an intermediate variable to avoid indenting:
```python
ModelFill = self.Model.with_context(fill_temporal=True)
groups = ModelFill.read_group([], fields=['date', 'value'], groupby=['date'])
```